azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -106,22 +124,82 @@ public int createImageType(final ImageType imageType) {
+      handleSqlException(e);
+    }
+    return imageTypeId;
+  }
+
+  @Override
+  public int addOwnerOfImageType(final ImageType imageType) {
+    final Set<ImageOwnership> currentOwners =
+        new HashSet<>(getImageTypeOwnership(imageType.getName()));
+    final Set<ImageOwnership> newOwners = new HashSet<>(imageType.getOwnerships());
+    newOwners.removeAll(currentOwners);
+    //Check if there are actually new owners added.
+    if (newOwners.isEmpty()) {
+      throw new ImageMgmtDaoException(ErrorCode.BAD_REQUEST, "Exception while updating image, "
+          + "specified no new owners");
+    }
+    int imageTypeId =
+        getImageTypeByName(imageType.getName()).map(BaseModel::getId).orElse(0);
+    final SQLTransaction<Integer> update = transOperator -> {
+      final Timestamp currentTimestamp = Timestamp.valueOf(LocalDateTime.now());
+      for (final ImageOwnership imageOwnership : newOwners) {
+        transOperator.update(INSERT_IMAGE_OWNERSHIP, imageTypeId, imageOwnership.getOwner(),
+            imageOwnership.getRole().name(), imageType.getCreatedBy(), currentTimestamp,
+            imageType.getModifiedBy(), currentTimestamp);
       }
-      if (e.getErrorCode() == SQL_ERROR_CODE_DATA_TOO_LONG) {
-        errorMessage = "Reason: Data too long for one or more column(s).";
+      transOperator.getConnection().commit();
+      return 1;
+    };
+    try {
+      this.databaseOperator.transaction(update);
+      if (imageTypeId < 1) {
+        log.error(String.format("Exception while updating image type due to invalid "
+            + "imageTypeId: %d.", imageTypeId));
+        throw new ImageMgmtDaoException(ErrorCode.BAD_REQUEST, "Exception while updating image ");
       }
-      throw new ImageMgmtDaoException(ErrorCode.BAD_REQUEST, "Exception occurred while creating "
-          + "image type metadata. " + errorMessage);
+      log.info("Successfully Updated image ownerships for :" + imageType.getName());
+    } catch (final SQLException e) {
+      log.error("Unable to update the image type metadata", e);
+      handleSqlException(e);
     }
     return imageTypeId;
   }
 
+  @Override
+  public int removeOwnerOfImageType(ImageType imageType) throws ImageMgmtException {
+    final Set<ImageOwnership> currentOwners =
+        new HashSet<>(getImageTypeOwnership(imageType.getName()));
+    final Set<ImageOwnership> ownersToRemove = new HashSet<>(imageType.getOwnerships());
+    ownersToRemove.retainAll(currentOwners);
+      //Check if it's removing all owners
+    if (ownersToRemove.size() >= currentOwners.size() - 1){
+        throw new ImageMgmtDaoException(ErrorCode.BAD_REQUEST, "Exception while deleting "
+            + " image owners, need greater than two owners to be present");
+      }
+      int imageTypeId =
+          getImageTypeByName(imageType.getName()).map(BaseModel::getId).orElse(0);
+
+    try {
+      if (imageTypeId < 1) {
```

Comment:
@djaiswal83 's comment above applies here too.

Reply:
It is re-arranged here right ? I presume this is the correct way ? where I check if the imageType id before the sql update.